pandas/core/groupby/generic.py

```diff
@@ -1101,7 +1108,8 @@ def _cython_agg_blocks(
                     deleted_items.append(locs)
                     continue
                 else:
-                    result = cast(DataFrame, result)
+                    # Cast to DataFrame-like type
+                    result = self.obj._constructor(result)
```

Comment:
this is not the same, is mypy complaining unless you do this?

Reply:
yes -- mypy was complaining about this usage:
```python
result = cast(type(self.obj), result)
```
Please let me know if you have an alternative suggestion.

Reply:
you can try
`assert isinstance(result, type(self.obj))`

cc @simonjayhawkins 

we don't want to actually call the constructor again